mypy.ini

```
@@ -0,0 +1,18 @@
+[mypy]
+files =
+  docs,
+  share,
+  src,
+  tests
+# FIXME: get rid of missing imports ignore
+ignore_missing_imports = true
```

Let's keep the trailing comma for better diffs + it's best to put comments on the same line if the length allows (otherwise something unrelated may appear between the comment line and the setting line over the course of refactoring).
```suggestion
  tests,
ignore_missing_imports = true  # FIXME: get rid of missing imports ignore
```

---

I'll add it back, but will have to remove it again after.

It will fail:

```
MyPy, for Python 3.6....................................................................................Failed
- hook id: mypy
- exit code: 2

Generated TXT report (via XSLT): /home/bthornto/github/ansible-navigator/.tox/.tmp/.mypy/index.txt
docs/_ext/single_sourced_data.py: error: Duplicate module named
"single_sourced_data" (also at "docs/_ext/single_sourced_data.py")
docs/_ext/single_sourced_data.py: note: Are you missing an __init__.py? Alternatively, consider using --exclude to avoid checking one of them.
Found 1 error in 1 file (errors prevented further checking)
```

---

Is it about the comma? Or the inline comment? Leave out one of them that breaks things and apply the other, I just needed to understand why...

---

With the comma mypy's effective configuration is:

```
 'files': ['docs', 'share', 'src', 'tests', ''],
```

without it is

```
 'files': ['docs', 'share', 'src', 'tests'],
```

I believe the empty entry causes my to run against the directories specified and everything, which results in 

```
docs/_ext/single_sourced_data.py: error: Duplicate module named "single_sourced_data" (also at "docs/_ext/single_sourced_data.py")
docs/_ext/single_sourced_data.py: note: Are you missing an __init__.py? Alternatively, consider using --exclude to avoid checking one of them.
```

---

Okay, good to know, so you're right to drop it. What about the inline comment, though?

---

Removing the comment, adding an empty line, on one line with a comma do not change the way it is parsed

---

> on one line with a comma do not change the way it is parsed

What about adding the comment to the line with the setting it refers to?

---

```
mypy.ini: [mypy]: ignore_missing_imports: Not a boolean: true  # FIXME: get rid of missing imports ignore
```

---

See below

---

Resolved.